src/main/java/ch/njol/skript/expressions/ExprBlocks.java

```diff
@@ -157,6 +158,8 @@ public Iterator<Block> iterator(Event event) {
 						if (number != null)
 							distance = number.intValue();
 					}
+				} else if (this.direction instanceof ExprFunctionCall) {
```

Comment:
This check is not the way: a function can return a direction e.g. `above`, and similarly a non-function expression may return a vector e.g. `vector from yaw 90 and pitch 0`